feat: add plan command

swaingotnochill opened the pull request:
This commit adds the plan command to the palette. Currently it runs the plan command with default flags on current working directory with help of an async spawn process.

swaingotnochill committed:
Add plan command

This commit adds the plan command to the pallette. Currently it runs the plan command with default plags on current working directory with help of an async spawn process.

removed unncessary png

removed unnecessary png file

Signed-off-by: Roshan Swain <swainroshan001@gmail.com>

HarikrishnanBalagopal commented on .gitignore:
```diff
@@ -0,0 +1 @@
+node_modules
```
Probably we can use this https://github.com/github/gitignore/blob/main/Node.gitignore

HarikrishnanBalagopal commented on out/extension.js.map:
```diff
@@ -0,0 +1 @@
+{"version":3,"file":"extension.js","sourceRoot":"","sources":["../src/extension.ts"],"names":[],"mappings":";;;AAAA,iCAAiC;AACjC,8DAA2D;AAC3D,yDAAuF;AAEvF,SAAgB,QAAQ,CAAC,OAAgC;IACvD,yBAAyB;IACzB,MAAM,aAAa,GAAG,MAAM,CAAC,MAAM,CAAC,mBAAmB,CAAC,WAAW,CAAC,CAAC;IACrE,IAAA,gDAAkC,EAAC,aAAa,CAAC,CAAC;IAElD,sCAAsC;IACtC,MAAM,qBAAqB,GAAG,MAAM,CAAC,QAAQ,CAAC,eAAe,CAAC,oBAAoB,EAAE,GAAG,EAAE;QACvF,iCAAe,CAA ...
```
We should .gitignore the files in the build directory called `out/`

HarikrishnanBalagopal commented on webview-ui/src/index.tsx:
```diff
@@ -0,0 +1,10 @@
+import ReactDOM from "react-dom";
+import App from "./App";
+
+ReactDOM.render(
```
We can use the newer React functions like `createRoot` https://reactjs.org/docs/rendering-elements.html#rendering-an-element-into-the-dom

HarikrishnanBalagopal commented on webview-ui/package.json:
```diff
@@ -0,0 +1,23 @@
+{
+  "name": "hello-world",
```
change the name to `move2kube-vscode-plugin`

HarikrishnanBalagopal commented on webview-ui/package.json:
```diff
@@ -0,0 +1,23 @@
+  },
+  "dependencies": {
+    "@vscode/webview-ui-toolkit": "^1.2.1",
+    "react": "^17.0.2",
```
Use the latest react 18

HarikrishnanBalagopal commented on webview-ui/package.json:
```diff
@@ -0,0 +1,23 @@
+  "dependencies": {
+    "@vscode/webview-ui-toolkit": "^1.2.1",
+    "react": "^17.0.2",
+    "react-dom": "^17.0.2"
```
use react-dom 18

HarikrishnanBalagopal commented on webview-ui/package.json:
```diff
@@ -0,0 +1,23 @@
+    "react-dom": "^17.0.2"
+  },
+  "devDependencies": {
+    "@types/react": "^17.0.33",
```
same as above

HarikrishnanBalagopal commented on webview-ui/package.json:
```diff
@@ -0,0 +1,23 @@
+  },
+  "devDependencies": {
+    "@types/react": "^17.0.33",
+    "@types/react-dom": "^17.0.10",
```
same as above

HarikrishnanBalagopal commented on out/extension.js:
```diff
@@ -0,0 +1,27 @@
+const HelloWorldPanel_1 = require("./panels/HelloWorldPanel");
+const getHandlers_1 = require("./utilities/getHandlers");
+function activate(context) {
+    // Create output channel.
```
The build output directory `out/` needs to be added to `.gitignore`

HarikrishnanBalagopal reviewed (changes requested):
Looks good, have left some comments. PTAL.

HarikrishnanBalagopal commented:
> This commit adds the plan command to the palette. Currently it runs the plan command with default flags on current working directory with help of an async spawn process.

I thought we were focussing on the `transform` command?

swaingotnochill commented:
> > This commit adds the plan command to the palette. Currently it runs the plan command with default flags on current working directory with help of an async spawn process.
> 
> I thought we were focussing on the `transform` command?

Since transform command needs plan, I thought this should be the first step towards it.

swaingotnochill committed:
Update the react modules to react 18

This commit aupdates all the node packages to their latest version, majorly moving react and react dom to version 18. Additionally, changed from single root ele element to createRoot. Update the gitignore to exclude build files.

Signed-off-by: Roshan Swain <swainroshan001@gmail.com>

HarikrishnanBalagopal commented:
> > > This commit adds the plan command to the palette. Currently it runs the plan command with default flags on current working directory with help of an async spawn process.
> > 
> > 
> > I thought we were focussing on the `transform` command?
> 
> Since transform command needs plan, I thought this should be the first step towards it.

No, we can use the single step flow which does not require plan phase.
```
$ move2kube transform -s ./path/to/src/folder
```

HarikrishnanBalagopal commented on package.json:
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "mov2kube-vscode-plugin",
```
typo

swaingotnochill replied:
Okay. I missed it. Sorry for this.

HarikrishnanBalagopal reviewed (changes requested):
Just a typo, otherwise looks good. Thanks for making the changes!

swaingotnochill committed:
Fix typo in package.json

Signed-off-by: Roshan Swain <swainroshan001@gmail.com>

swaingotnochill commented:
> > > > This commit adds the plan command to the palette. Currently it runs the plan command with default flags on current working directory with help of an async spawn process.
> > > 
> > > 
> > > I thought we were focussing on the `transform` command?
> > 
> > 
> > Since transform command needs plan, I thought this should be the first step towards it.
> 
> No, we can use the single step flow which does not require plan phase.
> 
> ```
> $ move2kube transform -s ./path/to/src/folder
> ```

Ohh. I will start to work on it.

HarikrishnanBalagopal commented:
> > > > > This commit adds the plan command to the palette. Currently it runs the plan command with default flags on current working directory with help of an async spawn process.
> > > > 
> > > > 
> > > > I thought we were focussing on the `transform` command?
> > > 
> > > 
> > > Since transform command needs plan, I thought this should be the first step towards it.
> > 
> > 
> > No, we can use the single step flow which does not require plan phase.
> > ```
> > $ move2kube transform -s ./path/to/src/folder
> > ```
> 
> Ohh. I will start to work on it.

Thanks, also as a side note, don't force push to the PR branch. It makes it very hard to see the update without reviewing the entire code again. Github already allows us to squash all the commits when the PR gets accepted and merged.

Also look at conventional commits for a standard way to write commit messages https://www.conventionalcommits.org/en/v1.0.0/#examples

HarikrishnanBalagopal reviewed (approved):
LGTM
